components/Dashboard/ItTable.vue

```diff
@@ -1,172 +1,347 @@
-
-
-
-
-
-        <el-table-column
-         width="100px"
-      label="Date"
-      prop="date">
-    </el-table-column>
-
-    <el-table-column
-     width="200px"
-      label="Status"
-     >
-       <el-tag
-         
-          disable-transitions >{{durum}}</el-tag>
-    </el-table-column>
-
-    
-
- <el-table-column
-     width="200px"
-      label="Explanation"
-      prop="explain">
-    </el-table-column>
-
-
-
-    
-    <el-table-column
-     width="300px"
-      align="right">
-      <template slot="header">
-        <el-input
-          v-model="search"
-          size="mini"
-          placeholder="Type to search"/>
-      </template>
-      <template >
-        <el-button
-          size="mini"
-            type="primary"
-           icon="el-icon-edit" >Edit</el-button>
-        <el-button
-          size="mini"
-          type="danger"
-         @click="dialogVisible = true" icon="el-icon-delete">Delete</el-button>
-      </template>
-    </el-table-column>
-  </el-table>
-
-
-  <el-dialog
-  title="Delete Operation"
-  :visible.sync="dialogVisible"
-  width="30%"
-  >
-  <span><b>Are you sure that you want to delete this record?</b></span>
-  <span slot="footer" class="dialog-footer">
-    <el-button @click="dialogVisible = false">Cancel</el-button>
-    <el-button type="success" @click="dialogVisible = false">Confirm</el-button>
-  </span>
-</el-dialog>
-</div>
+      >
+      <el-table-column align="center" label="#" type="index"></el-table-column>
+      <el-table-column
+        width="150"
+        style="height: 20%"
+        align="center"
+        label="Staff Name"
+        prop="name"
+      ></el-table-column>
+      <el-table-column width="200" align="center" label="Los" prop="los"></el-table-column>
+      <el-table-column width="100" align="center" label="Title" prop="titles"></el-table-column>
+      <el-table-column width="100" align="center" label="Location" prop="location"></el-table-column>
+      <el-table-column sortable width="150" align="center" label="Phone Model" prop="phoneModel"></el-table-column>
+      <el-table-column
+        sortable
+        width="200"
+        align="center"
+        label="Computer Model"
+        prop="computerModel"
+      ></el-table-column>
+      <el-table-column width="100" align="center" label="Date" prop="Hire_Date"></el-table-column>
+      <el-table-column width="150" align="center" label="Status" prop="process_status_short">
+        <el-tag disable-transitions>{{sizeFormT.process_status_short}}</el-tag>
+      </el-table-column>
+
+      <el-table-column style="height: 20%" align="center">
+        <template slot="header">
+          <el-input v-model="search" size="mini" placeholder="Type to search"/>
+        </template>
+
+        <template slot-scope="scope" style="width: 50% height:20%">
+          <el-button
+            size="mini"
+            type="success"
+            @click="dialogVisible = true;fetchDataFromTable(scope.$index,scope.row.id)"
```

Comment:
Move this logic to a function, it's hard to read this way.

Reply:
I created new function this problem.